### ISSUE TITLE
Is the repo working?

### ISSUE THREAD

**Issue by delphicgp** [andyswe/elva#1]
Hi Andy,

I am looking for a solution access the data from my KTC RCU1111 and parse the data into a nice diagram to display. I guess your repo is solving it. But when I read your code, I didn't quite understand how it shall work. I don't see how to use gradle to build your project. I don't find where is the place you define the ip address of your KTC meter.

Could you explain a little bit more?

Looking forward to your answer,
Gongpei

**Comment by andyswe** [comment]
Yeah, it is working. However i had no time to improve the design and code. The data from the KTC meter is pulled in Obtainer.kt. As you see the ip is hardcoded of now...  
You can make it configurable if you want, and then configure db in application.properties where data will be written.

**Comment by delphicgp** [comment]
Thanks for the quick answer. How do we compile it in gradle. I don't see any build task. 

Have you hard code any db of your own account?

**Comment by andyswe** [comment]
did you get it to work? dont know how tailored the KTC RCU unit is for every installation.

**Comment by delphicgp** [comment]
No, I didn't get it work. I had issue with spring in the gradle script. Also because i have never used kotlin before.

I find a frontend, home assistent. https://www.home-assistant.io/

If i can not get your code work, I will try that frontend. Then the only thing i need is from the web request,response, collect the information and report to the home assistant API.

I will try that later. I will let u know after i have a working solution.

By the way, before, I have KTC out of my home network. 

             fiber
           \ a switch?
   \\ KTC        \\Telia router
                            \\\ home network

The problem is i cannot access the KTC via ip address. I guess maybe the switch set different vlan of the ktc and telia port.

Now i move KTC into my home network. Then i can access the KTC page. Do you know if using this way, KTC can still get my energy report? What way do you install?

**Comment by andyswe** [comment]
i wouldnt move the KTC from what was the original installation, probably they cannot read energy reports then. We have huge problem with that in our building.

I dont really understand your setup. Do you have two LANs? My KTC just sits on the LAN after the switch/router and there are probably some firewall/forwarding/VLAN rules in the router, so it can be read from outside. I can access the KTC from my LAN. You have to sort that out.

My code should be buildable with "gradlew bootJar". If you are not ok with Kotlin, try out an online Kotlin-to-Java converter and go for Java. The KTC unit is very stupid implemented and is stateful in a way that make a simple curl/wget to retrive data difficult IMO.

**Comment by delphicgp** [comment]
After the fiber, there is a device. The device connects to KTC and my home router. But I cannot access KTC from my home network via the device, because KTC and my home router are in two different network and using different gateway. The only way i found is put the KTC into my home network, so it is in the network that i can access.

Could you tell me what exactly the curl/wget address?

I found if i visit http://192.168.1.65/mb.xml, i will get a xml. But the xml only include the data shows last in the http://192.168.1.65/mb.htm. So i don't know how to access data from MB01, MB02, MB03

**Comment by andyswe** [comment]
Basically you access that mb.htm page and set parameter 'counter' to iterate through all the MB pages until you get what you needed. This is my code snipper (kotlin)
`  var elDocument: Document? = null
        var vvDocument: Document? = null
        var counter = 0
        while (elDocument == null || vvDocument == null && counter < 10) {
            val document = obtainMb(httpClient)
            val mbName = document.selectSingleNode("/data/DBIpMB1_Txt").stringValue
            when (mbName) {
                "EL-01" -> elDocument = document
                "VV-01" -> vvDocument = document
            }
            counter++
        }
        val data =  Data(elDocument, vvDocument!!)
        return Parser().parse(data)
    }

    private fun obtainMb(httpClient: HttpClient): Document {




        val pageRequest = RequestBuilder.post()
                .setUri(URI("http://192.168.1.226/mb.xml"))
                    .addParameter("counter", "2")
                .addHeader("Content-Type", "text/xml")
                .build()

        val page = httpClient.execute(pageRequest)
        return reader.read(page.entity.content)
    }`

**Comment by delphicgp** [comment]
OK Understand. Thanks
Do you have a screen shoot of your frontend?

Do you know home assistant?

**Comment by andyswe** [comment]
![all](https://user-images.githubusercontent.com/17947979/51209827-79cb0f00-1911-11e9-8aa1-b2dd1e3a22c7.png)

I have minute granularity on electricity and water. this graphs shows monthly and weekly moving average on electricity.

**Comment by delphicgp** [comment]
Thank you for sharing knowledge.

I will try home assistant frontend. I will share you the result after i get it work.
